### ISSUE TITLE
Running `ember s` results in `file changed` for all files

### ISSUE THREAD

**Issue by Alonski** [ember-cli/ember-cli#9055]
Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 3.7.1
node: 12.15.0
v8: 7.7.299.13-node.16
uv: 1.33.1
zlib: 1.2.11
brotli: 1.0.7
ares: 1.15.0
modules: 72
nghttp2: 1.40.0
napi: 5
llhttp: 2.0.4
http_parser: 2.9.3
openssl: 1.1.1d
cldr: 35.1
icu: 64.2
tz: 2019c
unicode: 12.1
os: win32 x64
6.13.4
1.22.0
shell: Powershell Core
```

Running `ember s` I immediately get a `file changed` message for every file in my ember project...

Since we have a lot of files this takes quite some time. Eventually the project gets built and served correctly. 

Not sure how to start debugging this.

**Comment by rwjblue** [comment]
I've seen some form of this happen when something in the build pipeline writes to the input directories.

**Comment by Alonski** [comment]
Hmm, it doesn't happen on all machines all the time. It had has happened to teammates on and off over the last year. Yesterday it started happening to me every time I started the ember server.

Any ideas how to start debugging this?

**Comment by rwjblue** [comment]
Are you using watchman? What OS? Have you tried specifying alternate watchers (via the command line flag)? 

Sometimes FSEvents gets messed up and requires a restart to fix....

**Comment by Alonski** [comment]
@rwjblue I am not using watchmen. I am on Windows 10 using Powershell Core.

Should I try restarting the computer? Can I do something else?

**Comment by rwjblue** [comment]
Well, I'm not really sure. I think that there have been a number of changes to how files are watched since ember-cli@3.7 (which is what I think you said you were using). It is super hard to know if those are related or not, and it sounds like you can't reproduce at will (e.g. some machines are fine for a while, then start having an issue, etc).

**Comment by btecu** [comment]
I've seen this on Windows computers as well. I was not able to reproduce it on mine.

**Comment by theoriginaljaydubya** [comment]
I'm running into this as well on my work laptop.  It's running Windows 10.  My theory is virus protection (ESET).  I have not been able to validate it because corporate hasn't white-listed the repository.

**Comment by aluink** [comment]
I have this happening on my box, almost, all the time. It sometimes even triggers at random when `ember s` has been running for a while. For instance, sometimes I'll change a single file and the CLI notes that file changed and then while it's rebuilding, it spams tons of file changes and then rebuilds multiple times before it settles and finishes.  @theoriginaljaydubya and I happen to work together but I got our IT department to set an exclusion in my ESET for the entire repo where my multiple ember projects live.

To summarize, 

1. I see this in multiple ember projects
2. I see it both at start and in the middle of `ember s`'s lifespan
3. Virus protection (ESET) exclusion in place
4. It's been happening for a long time now, so a restart seems pointless now. I've rebooted a lot since.

**Comment by Alonski** [comment]
It stopped happening to me after happening consistently for more than a day. I will keep monitoring and trying to figure out why this happens.

**Comment by IBue** [comment]
I reproducibly run into this problem when upgrading from _ember-cli_ **3.12** to **3.16** (_Linux_, _watchman_).
For some reason, the build/serve process of 3.16 touches files (*.html, *.js) in `public/` and `bower_components`/ that are funneled into `dist/`, and therefore ends up in a perpetual rebuild loop.

Situation can be mitigated by adding affected source folders to the _watchman_ ignore list (`.watchmanconfig` → ignore_dirs; `watchman shutdown-server` required) at the cost of losing live-reload behaviour for these.

**Comment by Alonski** [comment]
Just happened to me again after restarting my laptop. :)
Again have no idea why and no idea how to debug

**Comment by Alonski** [comment]
Still happening :(
Anyone want to help me debug?

**Comment by Parrryy** [comment]
This is still happening for me as well. 
```
ember-cli: 3.8.3
node: 8.17.0
```

**Comment by kn1k** [comment]
What version of windows do you have? I encountered this problem after updating to 2004

**Comment by Alonski** [comment]
@kn1k Windows 10 2004?

**Comment by kn1k** [comment]
> @kn1k Windows 10 2004?

Yes

**Comment by KarlSchumann** [comment]
Just upgraded to Windows 10 2004 and am noticing this issue now as well. Anyone happen to come up with a solution?

**Comment by archaim** [comment]
Same here, Windows 10 2004

**Comment by Alonski** [comment]
So I have found that, at least for me, this is happening *only* when VSCode is open! 🙀 
It seems that VSCode gets a little to trigger-happy with watching/touching files.
Also when using VSCode search it sometimes triggers this again for all files. When changing excluded folders as well.

Not sure where to go from here 😄 

@archaim @KarlSchumann @kn1k Is this your case as well?

**Comment by jhsiaomei** [comment]
having the same issue on a mac
ember-cli: 3.2.0
node: 8.11.1
os: darwin x64

**Comment by archaim** [comment]
> @archaim @KarlSchumann @kn1k Is this your case as well?
No, it's still the same even without VSCode.

**Comment by tehmaestro** [comment]
Recently I've been having the same issue :(
Window 10, node 10, ember-cli 3.12, PHPStorm

**Comment by Jegadish** [comment]
Did anyone figure out what this issue is ? I can not find any other reports or forum posts of this happening anywhere and it makes my app's build time twice than it should be (locally).

**Comment by robertpbaxter** [comment]
Same issue on a mac, independent of editor. I have a bit of a standout difference though; this only occurs when I use `yarn link` when one of our repos is a dependency for another. I checked the file names and discovered that the node modules were being changed in the consuming app (not the linked one).
ember-cli: 3.4.4
node: 10.21
os: big sur 11.6
editor: neovim

**Comment by swaran18999** [comment]
This issue(for me at least) was being caused by node modules. 
A clean install of npm solves this issue for me. (npm ci && npm install).

**Comment by swaran18999** [comment]
This is a temporary workaround which I use. When this workaround is used, ember is rebuilt only when one particular file which we specify( **robots.txt** in the below code snippet) is saved.
**Workaround**: In the folder make the replace the makeOnChanged function in `<build_folder>/node_modules/ember-cli-broccoli-sane-watcher/index.js`
```
function makeOnChanged (log) {
  return function (filePath, root) {
    if (this.options.filter(path.basename(filePath), filePath, root)) {
      if (this.options.verbose) console.log(log, filePath);
      if(root.indexOf("node_modules")== -1 && filePath=="robots.txt"){
        console.log("rebuilding")
        this.scheduleBuild(path.join(root, filePath));
      }
    }
  };
}
```

**Comment by kumkanillam** [comment]
You need to add one if condition in the below files, 
node_modules\sane\src\node_watcher.js 
processChange method,

> //Allow only recently modified files, Since Windows receiving change event here when there is change in accessTime.             
            if(new Date()-stat.mtime<60000 || new Date()-stat.ctime<60000) { 
              this.emitEvent(CHANGE_EVENT, relativePath, stat);
            }else {
              console.log('Change event came, but not emitted: file',file);
            }

I modified in my fork for the reference.
[Link for reference](https://github.com/kumkanillam/sane/blob/33f4c37db8cdc2d4bdb282cc09455d09d56c071d/src/node_watcher.js#L308)
